golem-worker-service-base/src/evaluator/evaluator_context.rs

```diff
@@ -41,19 +77,27 @@ impl WorkerMetadataFetcher for NoopWorkerMetadataFetcher {
     async fn get_worker_metadata(
         &self,
         _worker_id: &WorkerId,
-    ) -> Result<Vec<AnalysedFunction>, MetadataFetchError> {
-        Ok(vec![])
+    ) -> Result<ComponentMetadata, MetadataFetchError> {
+        Ok(ComponentMetadata {
+            exports: vec![],
+            producers: vec![],
+        })
     }
 }
 
 impl EvaluationContext {
     pub fn empty() -> Self {
         EvaluationContext {
             variables: None,
-            analysed_functions: vec![],
+            functions: vec![],
         }
     }
 
+    pub fn find_function(&self, function: &str) -> Option<Function> {
```

Comment:
This is `O(n)` in number of functions, which is fine for now, but should change to `O(log n)` at some point.